src/main/java/com/shaft/cucumber/BrowserSteps.java

```diff
@@ -0,0 +1,109 @@
+package com.shaft.cucumber;
+
+import com.shaft.gui.browser.BrowserActions;
+import com.shaft.gui.browser.BrowserFactory;
+import com.shaft.validation.Assertions;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.openqa.selenium.WebDriver;
+
+public class BrowserSteps {
+    private final ThreadLocal<WebDriver> driver;
+
+    public BrowserSteps(ThreadLocal<WebDriver> driver) {
```

Comment:
Codacy found an issue: [Avoid reassigning parameters such as 'driver'](https://app.codacy.com/manual/mohab.mohieeldeen/SHAFT_ENGINE/pullRequest?prid=5731302)

Reply:
that's part of the dependency injection implementation